app/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationController.scala

```diff
@@ -82,13 +84,19 @@ class ProfitOrLossCalculationController @Inject() (override val messagesApi: Mes
       val ageIsTooYoung        = ageIsUnder16(userDoB, taxYear, ageAtStartOfTaxYear = false)
       val ageIsTooOld          = !ageIsUnderStatePensionAge(userDoB, taxYear, ageAtStartOfTaxYear = false)
 
-      (userIsClass2Eligible, ageIsTooYoung, ageIsTooOld) match { // TODO SASS-8740 add other cases for between class 2/4 thresholds
+      (userIsClass2Eligible, ageIsTooYoung, ageIsTooOld) match {
         case (true, true, _) => Some("tooYoung")
         case (true, _, true) => Some("tooOld")
         case _               => None
       }
     }
 
+  private def moreThanClass2ButLessThanClass4Threshold(taxYear: TaxYear, taxableProfitsAndLosses: List[TaxableProfitAndLoss])(implicit
+      request: DataRequest[_]): ApiResultT[Boolean] =
+    service.getUserDateOfBirth(request.nino, request.mtditid).map { _ =>
+      TaxableProfitAndLoss.class2EligibleButLessThenClass4Threshold(taxableProfitsAndLosses, taxYear)
+    }
```

Comment:
`service.getUserDateOfBirth` isn't being used so doesn't need to be ApiResultT.
Could we instead do
` def onPageLoad(taxYear: TaxYear, businessId: BusinessId): Action[AnyContent] = (identify andThen getData andThen requireData) async {
    implicit request =>
      val result = for {
        taxableProfitsAndLosses <- service.getAllBusinessesTaxableProfitAndLoss(taxYear, request.nino, request.mtditid)
        incomeSummary           <- service.getBusinessIncomeSourcesSummary(taxYear, request.nino, businessId, request.mtditid)
        netAmount          = incomeSummary.getNetBusinessProfitOrLossForTaxPurposes()
        profitOrLoss       = incomeSummary.returnProfitOrLoss()
        formattedNetAmount = formatSumMoneyNoNegative(List(netAmount))
        tables             = AdjustedTaxableProfitOrLossSummary.buildTables(taxYear, profitOrLoss)
        nicsExemptionMessage          <- showNicsExemptionMessage(taxYear, taxableProfitsAndLosses)
        showClass4AgeExemptionMessage <- showClass4AgeExemption(taxYear, taxableProfitsAndLosses)
      } yield Ok(
        view(
          request.userType,
          formattedNetAmount,
          taxYear,
          profitOrLoss,
          tables,
          nicsExemptionMessage,
          showClass4AgeExemptionMessage,
          journeys.routes.SectionCompletedStateController.onPageLoad(taxYear, businessId, ProfitOrLoss, NormalMode)
        )
      )
      handleResultT(result)
  }

  private def showNicsExemptionMessage(taxYear: TaxYear, taxableProfitsAndLosses: List[TaxableProfitAndLoss])(implicit
      request: DataRequest[_]): ApiResultT[Option[String]] =
    service.getUserDateOfBirth(request.nino, request.mtditid).map { userDoB =>
      val userIsClass2Eligible         = TaxableProfitAndLoss.areProfitsOrLossClass2Eligible(taxableProfitsAndLosses, taxYear)
      val ageIsTooYoung                = ageIsUnder16(userDoB, taxYear, ageAtStartOfTaxYear = false)
      val ageIsTooOld                  = !ageIsUnderStatePensionAge(userDoB, taxYear, ageAtStartOfTaxYear = false)
      val betweenEligibilityThresholds = TaxableProfitAndLoss.class2EligibleButLessThenClass4Threshold(taxableProfitsAndLosses, taxYear)

      (userIsClass2Eligible, ageIsTooYoung, ageIsTooOld, betweenEligibilityThresholds) match {
        case (true, true, _, _) => Some("class2Ineligible.tooYoung")
        case (true, _, true, _) => Some("class2Ineligible.tooOld")
        case (_, _, _, true)    => Some("betweenClass2AndClass4")
        case _                  => None
      }
    }`
    
    So we'd remove moreThanClass2ButLessThanClass4Threshold, and combine it into the pattern match

Reply:
Have dmed a formatted version